src/main/java/org/fluentd/logger/sender/AFUNIXSocketSender.java

```diff
@@ -0,0 +1,230 @@
+package org.fluentd.logger.sender;
+
+import org.fluentd.logger.errorhandler.ErrorHandler;
+import org.msgpack.MessagePack;
+import org.newsclub.net.unix.AFUNIXSocket;
+import org.newsclub.net.unix.AFUNIXSocketAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+@SuppressWarnings("Duplicates")
+public class AFUNIXSocketSender implements Sender {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AFUNIXSocketSender.class);
+
+    private static final ErrorHandler DEFAULT_ERROR_HANDLER = new ErrorHandler() {};
+
+    private MessagePack msgpack;
+
+    private AFUNIXSocketAddress server;
+
+    private AFUNIXSocket socket;
+
+    private int timeout;
+
+    private BufferedOutputStream out;
+
+    private ByteBuffer pendings;
+
+    private Reconnector reconnector;
+
+    private String name;
+
+    private ErrorHandler errorHandler = DEFAULT_ERROR_HANDLER;
+
+    // TODO:
+    public AFUNIXSocketSender(File socketFile, int port) {
+        this(socketFile, port, 3 * 1000, 8 * 1024 * 1024);
+    }
+
+    public AFUNIXSocketSender(File socketFile, int port, int timeout, int bufferCapacity) {
+        this(socketFile, port, timeout, bufferCapacity, new ExponentialDelayReconnector());
+    }
+
+    public AFUNIXSocketSender(File socketFile, int port, int timeout, int bufferCapacity, Reconnector reconnector) {
+        msgpack = new MessagePack();
+        msgpack.register(Event.class, Event.EventTemplate.INSTANCE);
+        pendings = ByteBuffer.allocate(bufferCapacity);
+
+        try {
+            server = new AFUNIXSocketAddress(socketFile, port);
+            this.reconnector = reconnector;
+            name = String.format("%s_%d_%d_%d", socketFile.toString(), port, timeout, bufferCapacity);
+            this.timeout = timeout;
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
This case is a critical situation. So, not only logging here (BTW, please use logger instead of printing an error message to STDOUT), throw the exception or check if `server` is null and throw a proper exception in `connect()`